lib/MusicBrainz/Server/Edit/Release/ReorderMediums.pm

```diff
@@ -96,8 +96,11 @@ sub build_display_data {
         sort { $a->{new} <=> $b->{new} }
         @{ $self->data->{medium_positions} } ];
 
-    $data{release} = $loaded->{Release}{ $self->data->{release}{id} }
-        || Release->new( name => $self->data->{release}{name} );
+    my $release = $loaded->{Release}{ $self->data->{release}{id} };
+    $self->c->model('ArtistCredit')->load($release);
+
+    $data{release} = $release || Release->new( name => $self->data->{release}{name} );
```

Comment:
I'd also just skip the `->load` invocation if release is undefined to begin with:
```suggestion
    if ($release) {
        $self->c->model('ArtistCredit')->load($release);
    } else {
        $data{release} = Release->new( name => $self->data->{release}{name} );
    }
```

Reply:
Well the `$data{release}` shouldn't be in the `else` or be changed, so ignore that part :)